eth/api.go

```diff
@@ -1158,14 +1161,29 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(txHash common.Hash) (ma
 
 	tx, _, err := getTransaction(s.chainDb, s.txPool, txHash)
 	if err != nil {
-		glog.V(logger.Debug).Infof("%v\n", err)
-		return nil, nil
+		glog.V(logger.Info).Infof("tzdybal 1 %v\n", err)
+		return nil, err
 	}
 
 	txBlock, blockIndex, index, err := getTransactionBlockData(s.chainDb, txHash)
 	if err != nil {
-		glog.V(logger.Debug).Infof("%v\n", err)
-		return nil, nil
+		glog.V(logger.Info).Infof("tzdybal 2 %v\n", err)
+		return nil, err
+	}
+
+	if replayTransactions && receipt.Status == types.TxStatusUnknown {
+		// TODO(tzdybal) - get rid of this nasty casting
+		proc := s.bc.Processor().(*core.StateProcessor)
+		statedb, err := s.bc.StateAt(s.bc.GetBlock(txBlock).Root())
```

Comment:
I think you may need to use the state from the parent block.

https://github.com/ethereumproject/go-ethereum/blob/master/core/blockchain.go#L1633